### PR TITLE
Feature/Graphical PRM modeling

### PR DESCRIPTION
**Pull request by bjoernmu**
Graphical PRM modeling with the topologymodeler

- Loading of Detector and Refinement Structure of an Pattern Refinement Model side by side into Topologymodeler.
- Provides the capability to draw Pattern Refinement Models as relationships.
- Add dropdown to select which Mapping Types should be displayed.

- [x] Ensure that you followed http://eclipse.github.io/winery/dev/ToolChain#github---prepare-pull-request.
- [x] Branch name checked. That means, the branch name starts with `thesis/`, `fix/`, ...
- [x] Ensure that the commit message is [a good commit message](https://github.com/joelparkerhenderson/git_commit_message)
- [x] Ensure to use auto format in **all** files
- [x] Ensure that you appear in `NOTICE` at Copyright Holders
- [ ] Tests created for changes
- [ ] Documentation updated (if needed)
- [ ] Screenshots added (for UI changes)

### CODE REVIEW

File: org.eclipse.winery.frontends/app/topologymodeler/src/app/canvas/canvas.component.ts [hunk 1]
```diff
@@ -97,7 +102,7 @@ export class CanvasComponent implements OnInit, OnDestroy, OnChanges, AfterViewI
     currentModalData: any;
     dragSourceActive = false;
     event: any;
-    selectedRelationshipType: EntityType;
+    selectedRelationshipType: EntityType = new EntityType('', '', '', '');
```
Comment (lharzenetter) [inline org.eclipse.winery.frontends/app/topologymodeler/src/app/canvas/canvas.component.ts:105]: Wouldn't it be easier to just use a default constructor?

File: org.eclipse.winery.frontends/package-lock.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "winery-frontends",
-    "version": "3.0.0-SNAPSHOT",
+    "version": "2.0.0-SNAPSHOT",
```
Comment (lharzenetter) [inline org.eclipse.winery.frontends/package-lock.json:3]: this looks like a wrong merge

File: org.eclipse.winery.frontends/package-lock.json [hunk 3]
```diff
@@ -6139,9 +6139,9 @@
             "integrity": "sha1-KujTxL7EyoxHbGhcQexKOkQUwtQ="
         },
         "ngx-bootstrap": {
-            "version": "3.0.0",
-            "resolved": "https://registry.npmjs.org/ngx-bootstrap/-/ngx-bootstrap-3.0.0.tgz",
-            "integrity": "sha512-Zu3m5Nwp6ISaqjrFeb1evFoaMn8EDR//RHoKeuLelvFOJjKHaCsbOdkWrAAtpTSapdc892mfZfWdTmSuaN2t5A=="
+            "version": "2.0.4",
```
Comment (lharzenetter) [inline org.eclipse.winery.frontends/package-lock.json:6142]: dito

File: org.eclipse.winery.frontends/package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "winery-frontends",
-    "version": "3.0.0-SNAPSHOT",
+    "version": "2.0.0-SNAPSHOT",
```
Comment (lharzenetter) [inline org.eclipse.winery.frontends/package.json:3]: see package.lock.json

**Review by lharzenetter — CHANGES_REQUESTED** [review]

**Comment by bjoernmu** [comment]
Follow up at https://github.com/eclipse/winery/pull/556

**Pull request closed without merging**